### PR TITLE
Automatically scroll on details menu when it opens off-screen

### PR DESCRIPTION
**Pull request by elenatanasoiu**
### What are you trying to accomplish?

If the details menu button is at the bottom of the page, we don't automatically scroll the page to show you the dropdown options. 

#### Example: dismissing code scanning alerts

When you click the "Dismiss alert" button in the code scanning alerts section, it will show you a list of three options in a dropdown.

However, when the "Dismiss alert" button is at the bottom of the page, we don't automatically scroll the page to show you the dropdown options. This leads to a bad experience as users might not notice there's a dropdown or they get a cut-off view of it:

Fix_input_to_action_by_aeisenberg_·_Pull_Request__1102_·_github_codeql-action_and_Google_Calendar_-_Week_of_June_19__2022

### What approach did you choose and why?

To fix this, let's listen for the `open` event on the dropdown menu and scroll the menu into view when it triggers.

I initially tried doing this by adding the `autofocus` property to the dropdown as the details component comes with this out-of-the-box. Unfortunately, that didn't work because that property can only be used with `input`, while we're using a radio group.

I then experimented with adding the `autofocus` to the first or last element in the radio button list (since it's actually an input) but the scrolling was not very smooth and it didn't really centre the dropdown menu in a pleasant way.

With `scrollIntoView`, the dropdown is always centred:

https://user-images.githubusercontent.com/1354439/176162175-73407e14-d696-41c1-b10f-1ebf34733854.mov

## Testing

I've added a unit test to illustrate how this should be used.

## Documentation

I've also added an example to the examples file.

## Feedback
Any feedback welcome!

## Anything else

This also addresses: https://github.com/github/details-menu-element/issues/58

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -137,6 +138,23 @@ function autofocus(details: Element): boolean {
...
+    const innerMenu = detailsMenu.closest('[role="menu"], [role="radiogroup"]')
+    if (innerMenu) {
+      innerMenu.scrollIntoView({behavior: 'smooth', block: 'center'})
+    } else {
+      return false
+    }
+    return true
+  } else {
+    return false
+  }
```
Comment (sampart) [inline src/index.ts:155]: Drive-by comment: Is it worth avoiding the nesting here?

```suggestion
  }
  return false
```

Comment (elenatanasoiu) [reply src/index.ts]: Thanks Sam, that's a good point! Have updated.

**Review by theinterned — COMMENTED** [review]
I worry that this interaction might be unexpected. Have you considered instead detecting when the menu is close to the bottom of the window, and opening upward instead?

This is what the native `select` does for example:

Screen Shot 2022-06-28 at 9 12 19 PM

Screenshot from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select

**Comment by theinterned** [comment]
Follow up, we have a behaviour that can be used for this https://github.com/primer/behaviors/blob/main/docs/anchored-position.md

**Comment by elenatanasoiu** [comment]
@theinterned Thanks for linking to that. Will have a proper read!
